### PR TITLE
error handling for deleted schemas

### PR DESCRIPTION
**Pull request by jkleung11**
- DE needed to start deleting some old versions of schemas since we were close to hitting a limit on Confluent Cloud
- Consumption of some events was dependent on some now deleted schema ids
- This PR is an attempt to add more error handling:
  - If the schema id exists: use it
  - If the schema id isn't valid, then use the latest schema for consumption

Any devs more well versed in PHP weighing in would be appreciated, not a skillset our team has unfortunately :/

### CODE REVIEW

File: KafkaPhp/Serializers/AvroSerializer.php [hunk 1]
```diff
@@ -91,7 +92,12 @@ private function nameFromEncodedPayload(string $payload): string
     {
         $schemaId = $this->getSchemaId($payload);
         /** @var \AvroRecordSchema $s */
-        $avroRecordSchema = $this->extractValueFromRegistryResponse($this->registry->schemaForId($schemaId));
+        try {
+            $avroRecordSchema = $this->extractValueFromRegistryResponse($this->registry->schemaForId($schemaId));
+        } catch(SchemaRegistryException $e) {
+            $subject = $this->getSubject($payload);
+            $avroRecordSchema = $this->extractValueFromRegistryResponse($this->registry->latestVersion($subject));
```
Comment (jasonvideoblocks) [inline KafkaPhp/Serializers/AvroSerializer.php:99]: Unless there's already a way to detect this condition, it feels like it'd be good to log that this is happening.  (If possible, not sure what environment this runs in.)

File: KafkaPhp/Serializers/AvroSerializer.php [hunk 2]
```diff
@@ -91,7 +92,12 @@ private function nameFromEncodedPayload(string $payload): string
     {
         $schemaId = $this->getSchemaId($payload);
         /** @var \AvroRecordSchema $s */
-        $avroRecordSchema = $this->extractValueFromRegistryResponse($this->registry->schemaForId($schemaId));
+        try {
+            $avroRecordSchema = $this->extractValueFromRegistryResponse($this->registry->schemaForId($schemaId));
+        } catch(SchemaRegistryException $e) {
```
Comment (jasonvideoblocks) [inline KafkaPhp/Serializers/AvroSerializer.php:97]: Is this something that we'd want to alert on, or at least log somewhere?  (If we don't already have a way to detect it.)

**Review by jasonvideoblocks — COMMENTED** [review]
I don't have any Kafka/Avro knowledge, buy it looks good from a PHP perspective.

**Comment by jasonvideoblocks** [comment]
Also I believe you can revert the changes to `composer.lock` since you didn't make any `composer.json` changes.

**Comment by jkleung11** [comment]
closed in favor or #33

**Pull request closed without merging**